examples/lapack/compile_time_dispatching/getrs_usm_mklcpu_cusolver.cpp

```diff
@@ -0,0 +1,331 @@
+        oneapi::mkl::backend_selector<oneapi::mkl::backend::cusolver>{ gpu_queue }, m, n, lda);
+    std::int64_t gpu_getrs_scratchpad_size = oneapi::mkl::lapack::getrs_scratchpad_size<float>(
+        oneapi::mkl::backend_selector<oneapi::mkl::backend::cusolver>{ gpu_queue },
+        trans, n, nrhs, lda, ldb);
+    float* gpu_getrf_scratchpad = sycl::malloc_device<float>(
+        gpu_getrf_scratchpad_size * sizeof(float), gpu_device, gpu_context);
+    float* gpu_getrs_scratchpad = sycl::malloc_device<float>(
+        gpu_getrs_scratchpad_size * sizeof(float), gpu_device, gpu_context);
+    if (!gpu_A || !gpu_B || !gpu_ipiv || !gpu_getrf_scratchpad) {
+        throw std::runtime_error("Failed to allocate USM memory.");
+    }
+
+    // copy data from host to CPU device
+    gpu_queue.memcpy(gpu_A, A.data(), A_size * sizeof(float)).wait();
+    gpu_queue.memcpy(gpu_B, B.data(), B_size * sizeof(float)).wait();
+
+    //
+    // Execute on CPU and GPU devices
+    //
+
+    cpu_getrf_done = oneapi::mkl::lapack::getrf(
+        oneapi::mkl::backend_selector<oneapi::mkl::backend::mklcpu>{ cpu_queue }, m, n, cpu_A, lda,
+        cpu_ipiv, cpu_getrf_scratchpad, cpu_getrf_scratchpad_size);
+    cpu_getrs_done = oneapi::mkl::lapack::getrs(
+        oneapi::mkl::backend_selector<oneapi::mkl::backend::mklcpu>{ cpu_queue },
+        trans, n, nrhs, cpu_A, lda, cpu_ipiv, cpu_B, ldb,
+        cpu_getrs_scratchpad, cpu_getrs_scratchpad_size, { cpu_getrf_done });
+    gpu_getrf_done = oneapi::mkl::lapack::getrf(
+        oneapi::mkl::backend_selector<oneapi::mkl::backend::cusolver>{ gpu_queue }, m, n, gpu_A,
+        lda, gpu_ipiv, gpu_getrf_scratchpad, gpu_getrf_scratchpad_size);
+    gpu_getrs_done = oneapi::mkl::lapack::getrs(
+        oneapi::mkl::backend_selector<oneapi::mkl::backend::cusolver>{ gpu_queue },
+        trans, n, nrhs, gpu_A, lda, gpu_ipiv, gpu_B, ldb,
+        gpu_getrs_scratchpad, gpu_getrs_scratchpad_size, { cpu_getrf_done });
+
+    // Wait until calculations are done
+    cpu_queue.wait_and_throw();
+    gpu_queue.wait_and_throw();
+
+    //
+    // Post Processing
+    //
+    // copy data from CPU device back to host
+    cpu_queue.memcpy(result_cpu.data(), cpu_B, B_size * sizeof(float)).wait_and_throw();
+
+    // copy data from GPU device back to host
+    gpu_queue.memcpy(result_gpu.data(), gpu_B, B_size * sizeof(float)).wait_and_throw();
+
+
+    // Print results
+    std::cout << "\n\t\tGETRF and GETRS parameters:" << std::endl;
+    std::cout << "\t\t\ttrans = "
+              << (trans == oneapi::mkl::transpose::nontrans
+                      ? "nontrans"
+                      : (trans == oneapi::mkl::transpose::trans ? "trans" : "conjtrans"))
+              << std::endl;
+    std::cout << "\t\t\tm = " << m << ", n = " << n << ", nrhs = " << nrhs << std::endl;
+    std::cout << "\t\t\tlda = " << lda << ", ldb = " << ldb << std::endl;
+
+    std::cout << "\n\t\tOutputting 2x2 block of A,B,X matrices:" << std::endl;
```

Comment:
technically is it only outputting A and X matrices?

Reply:
yes, removed B in latest commits. Since in the runtime examples we overwrite B with X, we only print A and X, but not B. Aligning here in compile time example to be consistent.